crypto/x509/x_x509a.c

```diff
@@ -125,14 +125,18 @@ int X509_add1_reject_object(X509 *x, const ASN1_OBJECT *obj)
 {
     X509_CERT_AUX *aux;
     ASN1_OBJECT *objtmp;
+    int ret;
     if ((objtmp = OBJ_dup(obj)) == NULL)
         return 0;
     if ((aux = aux_get(x)) == NULL)
         goto err;
     if (aux->reject == NULL
         && (aux->reject = sk_ASN1_OBJECT_new_null()) == NULL)
         goto err;
-    return sk_ASN1_OBJECT_push(aux->reject, objtmp);
+    ret = sk_ASN1_OBJECT_push(aux->reject, objtmp);
+    if (!ret)
```

Comment:
This test is wrong.  The code should be:
```c
if (ret > 0)
    return ret;
```

Reply:
My test is not wrong, only the other case. It is fine to change the code as you suggested.

Reply:
sk_ASN1_OBJECT_push() returns either 0 or -1 on error.  The -1 return isn't possible here but the proper check should still be used.

Reply:
thanks, agreed